probe_scraper/scrapers/git_scraper.py

```diff
@@ -71,13 +74,27 @@ def retrieve_files(repo_info, cache_dir):
 
                 results[_hash].append(disk_path)
                 timestamps[_hash] = ts
+
+        all_hashes = {}
+        for rel_path in repo_info.dependencies_files:
+            all_hashes.update(get_commits(repo, rel_path))
+        for _hash, ts in all_hashes.items():
+            if (min_date and ts < min_date):
+                continue
+
+            if repo_info.dependencies_format is not None:
+                repo.git.checkout(_hash)
```

Comment:
Do we really need this for all the hashes (e.g. can we shallow clone the 1 latest master commit?)? What happens if the application uses a component that publishes Glean metrics and then removes it 2 versions later?

Reply:
It runs gradle on every commit where the dependencies *may* have changed.  (Based on when the files specified in `dependencies_files` have changed, which includes certain key build.gradle files as well as Dependencies.kt.  This has already happened 204 times since Glean was integrated into Fenix.

This is done exactly for the case you mention -- if a dependency goes in and later comes out -- we still need to catch it and document it because it may have sent data at some point in the past.

It's possible we could get away with fewer checks on beta and release and just track release tags, or something.  But for nightly we're probably still stuck doing something like this.

Reply:
I see, thanks for clarifying. Looks like we're stuck with this, then :(

Reply:
We could cache these in s3, like we're doing with the nightly `histogram.json` files.

Reply:
@fbertsch: Can you point me at where that happens?  Is that the requests_cache [here](https://github.com/mozilla/probe-scraper/blob/master/probe_scraper/scrapers/moz_central_scraper.py#L219)?